src/java.base/share/classes/sun/net/www/MimeTable.java

```diff
@@ -155,22 +160,41 @@ public synchronized MimeEntry find(String type) {
      * with it. Parses general file names, and URLs.
      */
     public MimeEntry findByFileName(String fname) {
-        String ext = "";
+        // attempt to find the entry with the fragment component removed
+        MimeEntry entry = findByFileName(fname, true);
```

Comment:
I think we might need a check here first to see if the `fname` contains `#` and if it does, only then call the `findByFileName` with `true`. Without that check, with the change in this PR, it's now possible that the `findByFileName` will get called twice (once with `true` and once with `false`) for the case where the filename doesn't have a `#` and whose extension isn't in the MimeTable.

Reply:
Methods refactored in eab33e8000ea730c57c918dbf291af23bacbd059.